### ISSUE TITLE
Create an alpha layer to customize the vanilla Paper item

### ISSUE THREAD

**Issue by barrowisp** [barrowisp/craftastrophe#33]
Rather than recolor Mojang's item and use the result as our own Asset, make an image file that acts as an overlay, adding color as Minecraft combines the vanilla image with our image

#30

**Comment by barrowisp** [comment]
I tagged #33 in the associated Commit, but forgot to say "Resolves"

**Comment by yooksi** [comment]
Here are detailed steps on how to properly redo your commits so that they close issues. With this comment I will open this issue and if you follow the following steps it will be properly closed via reference.

-  creates a copy of our master branch that will be used to cherry-pick from later [§](https://git-scm.com/docs/git-checkout#Documentation/git-checkout.txt-emgitcheckoutemltbranchgt)
`$ git checkout -b master2`

- go back to master branch [§](https://git-scm.com/docs/git-checkout)
`$ git checkout master`

- remove last 3 commits from the master branch [§](https://git-scm.com/docs/git-reset)
`$ git reset HEAD~3`

- add all source files to the index so they can be removed with the reset command [§](https://git-scm.com/docs/git-add)
`$ git add src/*`

- remove all uncommited changes that are currently indexed [§](https://git-scm.com/docs/git-reset#Documentation/git-reset.txt---hard)
`$ git reset --hard`

- push and overwrite the origin branch tree with your local commit history [§](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force)
`$ git push --force`

Checkout the cloned branch `master2` and copy the commitSHA of the commit _just below the first one you want to cherry pick_ and the last commit. Past the two SHA codes somewhere where they can be easily accessed like a sticky note or a notepad file. Once you've done that go back to the original master branch (`checkout master`) and run a `cherry-pick` command like this:

`$ git cherry-pick <from this commit> <to this commit>`

![2019-05-13_202849](https://user-images.githubusercontent.com/16436920/57645337-557cf780-75be-11e9-9211-190e14202724.png)

Next you need to rebase your branch and reword the commits so they include the resolution keyword and the key of the issue they are resolving. First run a interactive rebase command up to the commit just below the last one you want to edit:

`$ git rebase -i <commit>`

Then instruct Git to reword your commits by replacing `pick` with `reword`:
```
reword 43534be Update binder.png
reword 8dc5801 Add semi-transparent overlays to the
pick a567cad Delegate method interception to AOPProxy
```
Fix commit description grammar and titles and add new line entries as list by prefixing them with a dash followed with a space like this: `- my list entry`. The resolution keyword should be included in a new line just below your description, see the last code block below.
```
Update binder texture

Replaced Binder texture with one sized a multiple of 16, as recommended in the debug log
```
```
Add blueprint semi-transparent overlay texture

- Add semi-transparent overlays to the vanilla Paper item and Book item, and designate the locations of those image files in the item model files. 
- However, the Binder doesn't look great using this overlay system; the edge flickers.
Resolves: #33
```
You're all done, just force push once again and the referenced issues will be closed once again. I know this is not an easy thing to do, that's why you have to make sure you don't make these kind of mistakes too often as it gets cumbersome to repeat this process. 

There is a simple Git rule that I would recommend you follow: "_commit often, push once...". Make sure all of your commits are properly formatted and include all the changes you want them to include before you do an actual push. This way we don't have to do force pushes that permanently alter the remote commit tree.
